### DISCUSSION TITLE
Question: VAD from Microphone Streaming using ONNX model

### DISCUSSION THREAD

**Discussion by Newo6174** [snakers4/silero-vad#72] [Q&A] [answered]
I have combined the code from ```Pyaudio Microphone Streaming Examples``` and ```ONNX VAD Example``` with a view to deploy the VAD on my raspberry pi. 

I replace the function ```validate()``` with ```validate_onnx```

However, I have encountered this bug and I am not sure how to solve it:
```[ONNXRuntimeError] : 2 : INVALID_ARGUMENT : Invalid rank for input: input Got: 1 Expected: 2 Please fix either the inputs or the model.```
How should I change the type of inputs (```audio_float32```) to solve it?
Thanks in advance.

**Comment by snakers4** [comment, marked as answer]
Please try checking the tensor size
From this error it seems that you are not feeding a 2D batch tensor, but just a 1D tensor

**Reply by Newo6174** [reply]
Add a dummy tensor to fix this issue ```unsqueeze(0)```
Thanks for the help

**Reply by snakers4** [reply]
In the end, is there an error with the utils, or was the error on your side?

**Reply by Newo6174** [reply]
If you want to use ONNX model in this [example](https://github.com/snakers4/silero-vad/blob/master/examples/pyaudio-streaming/pyaudio-streaming-examples.ipynb), then you need to add a dummy tensor.

**Reply by oumayma-ba** [reply]
How can i add a dummy tensor? please
